tests/selenium/test_scrolling.py

```diff
@@ -48,32 +68,64 @@ def test_scrol001_fixed_alignment(test, fixed_rows, fixed_columns, ops):
     fixed_width = test.driver.execute_script(
         "return parseFloat(getComputedStyle(document.querySelector('#table .cell-0-0')).width) || 0;"
     )
-    margin_left = test.driver.execute_script(
-        "return parseFloat(getComputedStyle(document.querySelector('#table .cell-0-1')).marginLeft);"
-    )
 
-    assert -margin_left == fixed_width
+    assert -get_margin(test) == fixed_width
+
+    scroll_by(test, 200)
 
-    test.driver.execute_script(
-        "document.querySelector('#table .row-1').scrollBy(200, 0);"
+    wait.until(
+        lambda: -get_margin(test) == fixed_width + 200, 3,
     )
 
+    scroll_by(test, -200)
+
     wait.until(
-        lambda: -test.driver.execute_script(
-            "return parseFloat(getComputedStyle(document.querySelector('#table .cell-0-1')).marginLeft);"
-        )
-        == fixed_width + 200,
-        3,
+        lambda: -get_margin(test) == fixed_width, 3,
     )
 
-    test.driver.execute_script(
-        "document.querySelector('#table .row-1').scrollBy(-200, 0);"
+
+@pytest.mark.parametrize(
+    "fixed_rows",
+    [dict(fixed_rows=dict(headers=True)), dict(fixed_rows=dict(headers=True, data=1))],
+)
+@pytest.mark.parametrize(
+    "fixed_columns",
+    [
+        dict(),
+        dict(fixed_columns=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True, data=1)),
+    ],
+)
+@pytest.mark.parametrize(
+    "ops", [dict(), dict(row_selectable="single", row_deletable=True)]
+)
+def test_scrol002_edit_navigate(test, fixed_rows, fixed_columns, ops):
+    props = {**base_props, **fixed_rows, **fixed_columns, **ops}
+
+    app = dash.Dash(__name__)
+    app.layout = DataTable(**props)
+
+    test.start_server(app)
+
+    target = test.table("table")
+    assert target.is_ready()
+
+    fixed_width = test.driver.execute_script(
+        "return parseFloat(getComputedStyle(document.querySelector('#table .cell-0-0')).width) || 0;"
     )
 
+    scroll_by(test, 200)
+
+    # alignment is ok after editing a cell
+    target.cell(0, 3).click()
+    test.send_keys("abc" + Keys.ENTER)
+
+    wait.until(lambda: -get_margin(test) == fixed_width + get_scroll(test), 3)
```

Comment:
This is presumably the same scroll position as the table had prior to the `send_keys` above, right? Is it necessarily the case that the table has finished rerendering after the `ENTER` before we execute this? Or do we need to look for something else that ensures rerendering is done? Same question re: the matching test below.

Reply:
I could add a wait for the `(0,4)` cell to be selected before checking the position to make 100% certain it re-rendered

Reply:
https://github.com/plotly/dash-table/pull/806/commits/315218f8bda067af01aae815fa8d1f0d7b588311